src/material/datepicker/datepicker-input-base.ts

```diff
@@ -344,9 +344,8 @@ export abstract class MatDatepickerInputBase<S, D = ExtractDateTypeFromSelection
 
   /** Formats a value and sets it on the input element. */
   protected _formatValue(value: D | null) {
-    this._elementRef.nativeElement.value = value
-      ? this._dateAdapter.format(value, this._dateFormats.display.dateInput)
-      : '';
+    this._elementRef.nativeElement.value =
+      value != null ? this._dateAdapter.format(value, this._dateFormats.display.dateInput) : '';
```

Comment:
Wouldn't think also try to format other falsy values like an empty string?

Reply:
Yep, it's already there. '!= null' checks only for 'null' and 'undefined'. 0, "", false, etc. will be formatted

Reply:
I think @crisbeto is saying that maybe `''` should be treated like `null`. I think it is technically correct to try to parse it though. We don't know what the generic type `D` will be, but it could be `string`. If its not `string` people shouldn't really be passing `''` anyways. I think we can keep it like this and try to presubmit it in Google. If we find a lot of people using `''` to mean `null` we may want to change it